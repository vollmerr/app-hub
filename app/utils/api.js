import { delay } from 'redux-saga';
import { call, put } from 'redux-saga/effects';
import decode from 'jwt-decode';
import 'whatwg-fetch';

import { authUserDone } from '../containers/AppHub/actions';


export const DEFAULT_EXPIRE = 1000;
export const TOKEN = 'id_token';


/**
 * Determines if should fetch based off a cache time
 *
 * @param {Date} date             - date of cache
 * @param {Number} cacheSeconds   - number of seconds to cache
 *
 * @return {Bool}                 - if should fetch
 */
export function shouldFetch(date, cacheSeconds = 350) {
  return !date || (new Date(date).getTime() - (cacheSeconds * 1000)) > new Date().getTime();
}


/**
 * Sets a token for authenication in localStorage
 *
 * @param {String} token  - token to set
 */
export function setToken(token) {
  localStorage.setItem(TOKEN, token);
}


/**
 * Gets the authenication token from localStorage
 * @return {String}     - authenication token
 */
export function getToken() {
  return localStorage.getItem(TOKEN);
}


/**
 * Clears the authenication token from localStorage
 */
export function clearToken() {
  localStorage.removeItem(TOKEN);
}


/**
 * Helper function to valdate that token exists and is not expired
 *
 * @param {object} token  - jwt token
 *
 * @return {bool}         - valid ? true : false
 */
export function validToken(token) {
  try {
    const { exp } = decode(token);
    const now = new Date().getTime().toString().substring(0, 10);

    return Number(exp) > Number(now);
  } catch (e) {
    return false;
  }
}


/**
 * Helper function to put jwt token into global redux store
 *
 * @param {String} token    - jwt token
 *
 * @return {Number}         - exipration time
 */
export function* putToken(token) {
  try {
    const {
      sub: sam,
      exp,
      iat,
      ...rest
    } = decode(token);

    const expire = (exp - iat) * 1000; // convert from ms to s

    yield put(authUserDone({ sam, expire, ...rest }));
    return expire;
  } catch (error) {
    throw new Error(error);
  }
}


/**
 * Authenticates a user
 * Checks local storage for token first
 *
 * @param {Object} action   - redux action to dispatch (must have 'type')
 * @param {Number} maxTries - max tries to re attempt authenicating on fail
 */
export function* authenticate(action, maxTries = 3) {
  let expire = DEFAULT_EXPIRE; // try again in 1s by default
  let tries = 0;
  let error = new Error('Failed to authenticate');
  while (tries < maxTries) { // eslint-disable-line
    const token = getToken();
    // if valid local token use that, otherwise get new token
    if (validToken(token)) {
      // update user in global appHub state
      expire = yield call(putToken, token);
      // reset max tries
      tries = 0;
    } else {
      try {
        const options = {
          method: 'get',
          credentials: 'include',
        };
        const { id_token } = yield call(request, API.JWT, options); // eslint-disable-line
        // set into local storage for future authentication caching
        setToken(id_token);
        // update user in global appHub state
        expire = yield call(putToken, id_token);
        // reset max tries
        tries = 0;
      } catch (err) {
        // increment max tries
        tries += 1;
        error = err;
      }
      // max number of tires, set error
      if (tries === maxTries) {
        yield put(authUserDone(error));
      }
    }
    // authenicate again when token has expired
    yield call(delay, expire);
  }
}


/**
 * Parses the JSON returned by a network request
 *
 * @param  {object} response A response from a network request
 *
 * @return {object}          The parsed JSON from the request
 */
export function parseJSON(response) {
  if (response.status === 204 || response.status === 205) {
    return null;
  }
  return response.json();
}


/**
 * Checks if a network request came back fine, and throws an error if not
 *
 * @param  {object} response   A response from a network request
 *
 * @return {object|undefined} Returns either the response, or throws an error
 */
export function checkStatus(response) {
  if (response.status >= 200 && response.status < 300) {
    return response;
  }

  const error = new Error(response.statusText);
  error.response = response;
  throw error;
}


/**
 * Requests a URL, returning a promise
 *
 * @param  {string} url       The URL we want to request
 * @param  {object} [options] The options we want to pass to "fetch"
 *
 * @return {object}           The response data
 */
export function request(url, options) {
  return fetch(url, options)
    .then(checkStatus)
    .then(parseJSON);
}


/**
 * Requests a URL with jwt token in header for authorization, returning a promise
 * Waits until jwt token is set in localStorage before making call
 *
 * @param  {string} url       The URL we want to request
 * @param  {object} [options] The options we want to pass to "fetch"
 *
 * @return {object}           The response data
 */
export function requestWithToken(url, options = {}) {
  let timer = null;

  const requestIfToken = () => { // eslint-disable-line
    // get users jwt token form storage else they dont have one
    const token = localStorage.getItem('id_token') || null;
    // valid token, call api
    if (validToken(token)) {
      clearTimeout(timer);
      // config settings to send api
      const newOptions = {
        method: options.method || 'get',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${token}`,
        },
        ...options,
      };
      // append optional body to request (POSTs)
      if (options.body) {
        newOptions.body = JSON.stringify(options.body);
      }

      return request(url, newOptions);
    }
    // wait again until token set
    timer = setTimeout(requestIfToken, 100);
  };

  return requestIfToken();
}
