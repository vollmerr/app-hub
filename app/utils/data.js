/**
 * Formats an item object to the correct format
 * for rendering
 *
 * @param {array} item      - item to format
 * @param {string} name     - internal field name for lookup
 * @param {object} field    - field format info
 * @param {object} enums    - display names for enums
 *
 * @return {string}         - formatted item
 */
export function formatItem(item, name, field, enums = {}) {
  let content = item[name];
  // convert all data to array for mapping over
  if (!Array.isArray(content)) {
    content = [content];
  }
  // go through each item, mapping to the correct format
  content = content.map((x) => {
    // is enum mapping
    if (enums[name]) {
      return enums[name][x];
    }
    // is date
    if (field.data && field.data.type === 'DATE') {
      return isNaN(Date.parse(x)) ? '' : new Date(x).toISOString().substr(0, 10);
    }
    return x;
  });
  // join them back as string seperated by commas
  return content.join(', ');
}

/**
 * Formats an array of item objects to the correct format
 * for rendering
 *
 * @param {array} items     - items to format
 * @param {object} fields   - lookup table of fields
 * @param {object} enums    - display names for enums
 *
 * @return {array}          - formatted items as objects of strings
 */
export function formatItems(items, fields, enums) {
  const formattedItems = [];

  items.forEach((item, i) => {
    formattedItems[i] = {};
    Object.keys(item).forEach((k) => {
      formattedItems[i][k] = formatItem(item, k, fields[k], enums);
    });
  });

  return formattedItems;
}

/**
 * Maps a object to the format Office-UI-Fabric-React DetailsList`s expect
 *
 * @param {object} obj    - object to map
 * @param {array} include - keys to include, if empty includes all
 * @param {array} exclude - keys to exclude
 *
 * @return {object}       - mapped object to use in List
 */
export function mapToColumns(obj, include = [], exclude = []) {
  const keys = include.length ?
    include :
    Object.keys(obj);

  return keys
    .filter((key) => !exclude.includes(key))
    .map((key) => ({
      key: obj[key].name,
      name: obj[key].label,
      fieldName: obj[key].name,
      isResizable: true,
      ariaLabel: obj[key].ariaLabel,
      minWidth: obj[key].minWidth || 100,
      maxWidth: obj[key].maxWidth || 200,
      data: obj[key].data || {},
    }));
}
