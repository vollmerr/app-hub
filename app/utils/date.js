export const dateWithoutTime = (date) => {
  if (date && !isNaN(new Date(date))) {
    const d = new Date(date);
    d.setHours(0, 0, 0, 0);
    return d;
  }
  return undefined;
};
