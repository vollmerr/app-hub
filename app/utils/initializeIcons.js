import { registerIcons } from '@uifabric/styling/lib/index';
import appHubIcons from 'images/office365icons.woff';

// http://o365icons.cloudapp.net/
const initializeIcons = () => {
  registerIcons({
    style: {
      MozOsxFontSmoothing: 'grayscale',
      WebkitFontSmoothing: 'antialiased',
      fontStyle: 'normal',
      fontWeight: 'normal',
      speak: 'none',
    },
    fontFace: {
      fontFamily: `"appHubIcons"`, // eslint-disable-line
      src: `url(${appHubIcons}) format('woff')`,
    },
    icons: {
      // NAV
      helpMenu: '\uE006',
      alertsMenu: '\uE010',
      appMenu: '\uE020',
      settingsMenu: '\uE035',
      appHubMenu: '\uE303',
      // __APPS__
      ED: '\uE417',
      Loading: '\uF8FD',
      Demo: '\uF8FD',
      // SPA: '\uE22E',
      Apphub: '\uE300',
      COLD: '\uE408',
      BARS: '\uE259',
      // requried by office-ui
      chevrondown: '\uE088',
      chevronup: '\uE087',
      Search: '\uE039',
      Clear: '\uE353',
      CircleRing: '\uF8FE',
      StatusCircleCheckmark: '\uF8FF',
      sortup: '\uF8FB',
      sortdown: '\uF8FC',
      calendar: '\uE38E',
      up: '\uE27A',
      down: '\uE27B',
      checkmark: '\uF8FA',
    },
  });
};

export default initializeIcons;
