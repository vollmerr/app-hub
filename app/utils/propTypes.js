import PropTypes from 'prop-types';

const { string, bool, shape, func, arrayOf } = PropTypes;

export const metaDataProp = shape({
  title: string,
  desc: string,
  keywords: string,
});

export const routeProp = shape({
  key: string,
  href: string,
  path: string,
  name: string,
});
export const routesProp = arrayOf(routeProp);

// REDUX-FORM
export const inputProp = (value) => shape({
  value,
  onChange: func,
  onBlur: func,
  onFocus: func,
});

export const metaProp = shape({
  error: string,
  touched: bool,
});

export const optionProp = shape({
  key: string,
  text: string,
});
export const optionsProp = arrayOf(optionProp);

