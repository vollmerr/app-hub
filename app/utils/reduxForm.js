export const mapOnBlur = (onBlur, input) => (event, newValue, previousValue) => {
  if (onBlur) {
    onBlur(event, newValue, previousValue);
  }
  input.onBlur(event, newValue, previousValue);
};

export const mapOnFocus = (onFocus, input) => (event) => {
  if (onFocus) {
    onFocus(event);
  }
  input.onFocus(event);
};

export const mapOnChange = (onChange, input) => (event, newValue, previousValue) => {
  if (onChange) {
    onChange(event, newValue, previousValue);
  }
  input.onChange(event, newValue, previousValue);
};
