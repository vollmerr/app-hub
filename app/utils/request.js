import 'whatwg-fetch';


/**
 * Downloads a file
 *
 * @param  {object} file    - File to download
 * @param  {string} name    - Name to save file as
 */
export /* istanbul ignore next */ function downloadFile(file, name) {
  // make into blob and save
  const a = document.createElement('a');
  const data = [file];
  const blob = new Blob(data, { type: 'octet/stream' });

  document.body.appendChild(a);
  // IE...
  if (window.navigator.msSaveOrOpenBlob) {
    window.navigator.msSaveOrOpenBlob(blob, name);
  } else {
    const url = window.URL.createObjectURL(blob);

    a.href = url;
    a.download = name;
    a.click();
    window.URL.revokeObjectURL(url);
  }
}


/**
 * Maps an array of objects into { byId: [], allIds: {} }
 *
 * @param {array} arr       - array to normalize
 * @param {string} key      - key to use identifier
 *
 * @return {object} byId    - mapped items by id
 * @return {array} allIds   - ids of all objects, in order
 */
export const normalizeById = (arr, key = 'id') => {
  const byId = arr.reduce((acc, cur) => {
    acc[String(cur[key])] = cur;
    return acc;
  }, {});
  const allIds = Object.keys(byId);
  return { byId, allIds };
};


/**
 * Parses the JSON returned by a network request
 *
 * @param  {object} response A response from a network request
 *
 * @return {object}          The parsed JSON from the request
 */
export function parseJSON(response) {
  if (response.status === 204 || response.status === 205) {
    return null;
  }
  return response.json();
}


/**
 * Checks if a network request came back fine, and throws an error if not
 *
 * @param  {object} response   A response from a network request
 *
 * @return {object|undefined} Returns either the response, or throws an error
 */
export function checkStatus(response) {
  if (response.status >= 200 && response.status < 300) {
    return response;
  }

  const error = new Error(response.statusText);
  error.response = response;
  throw error;
}


/**
 * Requests a URL, returning a promise
 *
 * @param  {string} url       The URL we want to request
 * @param  {object} [options] The options we want to pass to "fetch"
 *
 * @return {object}           The response data
 */
function request(url, options) {
  return fetch(url, options)
    .then(checkStatus)
    .then(parseJSON);
}

export default request;
