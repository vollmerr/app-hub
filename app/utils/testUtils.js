import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

/**
 * Tests a styled-component for basic tests
 * such as rending correctly and adopting attributes
 * @param {Object} Component    - React component under test
 * @param {Object} extend       - React component the component under test extends /  inherits from
 */
export function testStyledComponent(Component, extend = null) {
  describe(`<${Component.displayName} />`, () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Component />);
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it(`should render an <${Component.target}> tag`, () => {
      expect(wrapper.type()).toEqual(Component.target);
    });

    it('should have a className attribute', () => {
      expect(wrapper.prop('className')).toBeDefined();
    });

    it('should adopt a valid attribute', () => {
      const id = 'test';
      wrapper.setProps({ id });
      expect(wrapper.prop('id')).toEqual(id);
    });

    if (extend) {
      it(`should extend ${extend.name}`, () => {
        expect(wrapper.find(extend).exists()).toEqual(true);
      });
    } else {
      it('should not adopt an invalid attribute', () => {
        const attribute = 'test';
        wrapper.setProps({ attribute });
        expect(wrapper.prop('attribute')).toBeUndefined();
      });
    }
  });
}

/**
 * Tests a components for basic mapDispatchToProps information
 * @param {func} mapDispatchToProps   - components mapDispatchToProps function
 * @param {object} actions            - functions in mapDispatchToProps to test
 */
export function testMapDispatchToProps(mapDispatchToProps, actions) {
  const onAction = (action) => (
    `on${action.charAt(0).toUpperCase()}${action.slice(1)}`
  );

  let dispatch;
  let dispatchProps;
  beforeEach(() => {
    dispatch = jest.fn();
    dispatchProps = mapDispatchToProps(dispatch);
  });

  Object.keys(actions).forEach((action) => {
    describe(onAction(action), () => {
      it('should be injected', () => {
        expect(dispatchProps[onAction(action)]).toBeDefined();
      });

      it('should dispatch when called', () => {
        dispatchProps[onAction(action)]();
        expect(dispatch).toHaveBeenCalled();
      });
    });
  });
}
