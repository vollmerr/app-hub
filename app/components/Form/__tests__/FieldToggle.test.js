import React from 'react';
import { shallow } from 'enzyme';
import { Toggle } from 'office-ui-fabric-react/lib/Toggle';

import { testStyledComponent } from '../../../utils/testUtils';

import Default, { FieldToggle, StyledToggle } from '../FieldToggle';
import { FieldToggle as Index } from '../index';


testStyledComponent(StyledToggle, Toggle);


const styledToggleProps = {
  warning: false,
  disabled: false,
  required: false,
  checked: false,
  readOnly: false,
  errorMessage: null,
};


describe('StyledToggle', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<StyledToggle {...styledToggleProps} />);
  });

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly when checked', () => {
    wrapper.setProps({ checked: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly when is readOnly', () => {
    wrapper.setProps({ readOnly: true });
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ checked: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly when is disabled', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly when is required', () => {
    wrapper.setProps({ required: true });
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly when has warning', () => {
    wrapper.setProps({ warning: true });
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ checked: null });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly when has errorMessage', () => {
    wrapper.setProps({ errorMessage: 'error!' });
    expect(wrapper).toMatchSnapshot();
  });
});


const props = {
  meta: {
    error: null,
    touched: false,
  },
  input: {
    value: 0,
    onBlur: jest.fn(),
    onFocus: jest.fn(),
    onChange: jest.fn(),
  },
  isNullable: false,
};


describe('FieldToggle', () => {
  let wrapper;
  let instance;
  beforeEach(() => {
    wrapper = shallow(<FieldToggle {...props} />);
    instance = wrapper.instance();
  });

  it('should have a format property that coerces to a Number, and defaults to undefined', () => {
    expect(FieldToggle.format(undefined)).toEqual(undefined);
    expect(FieldToggle.format('1')).toEqual(1);
    expect(FieldToggle.format(0)).toEqual(0);
  });

  it('should be exported (wrapped) in the index', () => {
    expect(Default).toBe(Index);
  });

  describe('componentWillReceiveProps', () => {
    it('should update if `checked`', () => {
      const value = 1;
      instance.componentWillReceiveProps({ input: { value }, meta: {} });
      expect(wrapper.state('checked')).toEqual(value);
    });

    it('should update the `errorMessage`', () => {
      const error = 'test error';
      const meta = { error, touched: true };
      instance.componentWillReceiveProps({ input: {}, meta });
      expect(wrapper.state('errorMessage')).toEqual(error);
    });
  });


  describe('getChecked', () => {
    it('should return the value as a number', () => {
      expect(instance.getChecked(true)).toEqual(1);
      expect(instance.getChecked(1)).toEqual(1);
    });

    it('should return null and undefined values as 0', () => {
      expect(instance.getChecked(null)).toEqual(0);
      expect(instance.getChecked(undefined)).toEqual(0);
    });

    it('should return null and undefined values as undefined if `isNullable`', () => {
      wrapper.setProps({ isNullable: true });
      expect(instance.getChecked(null)).toEqual(undefined);
      expect(instance.getChecked(undefined)).toEqual(undefined);
    });
  });


  describe('handleChange', () => {
    it('should update redux form', () => {
      const value = 0;
      instance.handleChange(value);
      expect(props.input.onChange).toHaveBeenCalledWith(value);
    });
  });


  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a `StyledToggle`', () => {
      expect(wrapper.find(StyledToggle).length).toEqual(1);
    });

    it('should handle errors if touched', () => {
      const error = 'test error';
      const meta = { error, touched: true };
      wrapper.setProps({ meta });
      expect(wrapper.find(StyledToggle).prop('errorMessage')).toEqual(error);
    });

    it('should not render errors if not touched or no error', () => {
      let meta = { error: 'test error' };
      wrapper.setProps({ meta });
      expect(wrapper.find(StyledToggle).prop('errorMessage')).toEqual('');

      meta = { touched: true, error: null };
      wrapper.setProps({ meta });
      expect(wrapper.find(StyledToggle).prop('errorMessage')).toEqual('');
    });
  });
});
