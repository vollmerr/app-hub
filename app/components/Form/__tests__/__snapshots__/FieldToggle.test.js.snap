// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldToggle__StyledToggle /> should render correctly 1`] = `
.c0 .ms-Toggle-background + .ms-Label {
  visibility: hidden;
}

.c0 .ms-Toggle-background > * {
  visibility: hidden;
}

<CustomizedToggle
  className="c0"
/>
`;

exports[`FieldToggle render should render correctly 1`] = `
<FieldToggle__StyledToggle
  checked={0}
  errorMessage=""
  isNullable={false}
  meta={
    Object {
      "error": null,
      "touched": false,
    }
  }
  onChanged={[Function]}
/>
`;

exports[`StyledToggle should render properly 1`] = `
<CustomizedToggle
  checked={false}
  className=""
  disabled={false}
  errorMessage={null}
  readOnly={false}
  required={false}
  warning={false}
/>
`;

exports[`StyledToggle should render properly when checked 1`] = `
<CustomizedToggle
  checked={true}
  className=""
  disabled={false}
  errorMessage={null}
  readOnly={false}
  required={false}
  warning={false}
/>
`;

exports[`StyledToggle should render properly when has errorMessage 1`] = `
.c0 .ms-Toggle-background {
  background: #fde7e9;
  border-color: #a80000;
}

<CustomizedToggle
  checked={false}
  className="c0"
  disabled={false}
  errorMessage="error!"
  readOnly={false}
  required={false}
  warning={false}
/>
`;

exports[`StyledToggle should render properly when has warning 1`] = `
<CustomizedToggle
  checked={false}
  className=""
  disabled={false}
  errorMessage={null}
  readOnly={false}
  required={false}
  warning={true}
/>
`;

exports[`StyledToggle should render properly when has warning 2`] = `
.c0 .ms-Toggle-background {
  background: #fff4ce;
  border-color: #ffb900;
}

.c0 .ms-Toggle-background + .ms-Label {
  visibility: hidden;
}

.c0 .ms-Toggle-background > * {
  visibility: hidden;
}

<CustomizedToggle
  checked={null}
  className="c0"
  disabled={false}
  errorMessage={null}
  readOnly={false}
  required={false}
  warning={true}
/>
`;

exports[`StyledToggle should render properly when is disabled 1`] = `
<CustomizedToggle
  checked={false}
  className=""
  disabled={true}
  errorMessage={null}
  readOnly={false}
  required={false}
  warning={false}
/>
`;

exports[`StyledToggle should render properly when is readOnly 1`] = `
.c0 .ms-Toggle-background {
  background: #ffffff;
}

.c0 .ms-Toggle-thumb {
  background: #333333;
  border-color: #333333;
}

.c0 .ms-Label {
  color: #333333;
}

<CustomizedToggle
  checked={false}
  className="c0"
  disabled={false}
  errorMessage={null}
  readOnly={true}
  required={false}
  warning={false}
/>
`;

exports[`StyledToggle should render properly when is readOnly 2`] = `
.c0 .ms-Toggle-background {
  background: #0078d7;
}

.c0 .ms-Toggle-thumb {
  background: #ffffff;
  border-color: transparent;
}

.c0 .ms-Label {
  color: #333333;
}

<CustomizedToggle
  checked={true}
  className="c0"
  disabled={false}
  errorMessage={null}
  readOnly={true}
  required={false}
  warning={false}
/>
`;

exports[`StyledToggle should render properly when is required 1`] = `
.c0 > .ms-Label::after {
  content: " *";
  color: #a80000;
  padding-right: 12px;
}

<CustomizedToggle
  checked={false}
  className="c0"
  disabled={false}
  errorMessage={null}
  readOnly={false}
  required={true}
  warning={false}
/>
`;

exports[`StyledToggle should render properly when is required 2`] = `
<CustomizedToggle
  checked={false}
  className=""
  disabled={true}
  errorMessage={null}
  readOnly={false}
  required={true}
  warning={false}
/>
`;
